### PR TITLE
Code Clean Up, Remove deprecated Google Analytics (GA3) References

### PR DESCRIPTION
**Pull request by briowill**
Related to # https://github.com/Automattic/wp-calypso/pull/90481 https://github.com/Automattic/martech/issues/2318

## Proposed Changes
Google Universal Analytics (GA3 UA) has been deprecated and we have updated to Google Analytics 4.
This PR Continues the code clean up of removing unused GA3 UA code started in https://github.com/Automattic/wp-calypso/pull/90481

1. Remove deprecated  GA Enhanced Eccommerce
2. Remove unit tests under test/google-analytics
3. Removing deprecated ad-tracking/google-analytics
4. Refactor code to refer to ad-tracking/google-analytics-4 instead of ad-tracking/google-analytics

## Testing Instructions

1. Check out the branch
2. Allow ad-tracking and the GDPR banner in development.json
3. Open Calypso as a logged-in user.
4. You should see GA4 pixel being loaded, and the page_view and user_engagement events should be sent. 
5. Visit another page and the page_view event should be sent.
6. Sandbox the API and purchas a plan. 
7. You should see GA4 conversions being sent.

### CODE REVIEW

**Comment by briowill** [comment]
@gmovr  and @robertsreberski  Could you give this a look. It is the follow up to https://github.com/Automattic/wp-calypso/pull/90481

File: client/lib/analytics/ga.js [hunk 1]
```diff
@@ -58,18 +53,23 @@ export const gaRecordPageView = makeGoogleAnalyticsTrackingFunction( function re
 			useJetpackGoogleAnalytics +
 			'] [useAksiemtGoogleAnalytics: ' +
 			useAkismetGoogleAnalytics +
-			'] [useA8CForAgenciesGoogleAnalytics: ' +
```
Comment (gmovr) [inline client/lib/analytics/ga.js]: Is there a reson we remove this?

File: client/lib/analytics/ad-tracking/index.js [hunk 2]
```diff
@@ -1,12 +1,6 @@
 export { adTrackRegistration } from './ad-track-registration';
 export { adTrackSignupComplete } from './ad-track-signup-complete';
 export { adTrackSignupStart } from './ad-track-signup-start';
-export {
```
Comment (gmovr) [inline client/lib/analytics/ad-tracking/index.js:4]: Is there a reason these are moved one level up to the root instead of being in ad-tracking?

**Review by gmovr — COMMENTED** [review]
Hey! Could you leave testing instructions for next time? It will make it speedier to leave a review. :) 

**I tested it this way:**
- Checked out the branch locally, and changed `development.json` to allow ad-tracking and the GDPR banner.
- Opened Calypso as a logged-in user. I can see the GA4 pixel being loaded, and `page_view` and `user_engagement` events are sent. Switching the page to another one will kick off the `page_view` event as expected.
- Sanboxed the API and purchased a plan. I see the conversions as earlier in GA4.

File: client/lib/analytics/ga.js (hunk 1 above)
Comment (briowill) [reply client/lib/analytics/ga.js]: This wasn't intentional, seemed to happen when rebasing. I had manually added useA8CForAgenciesGoogleAnalytics as a function param locally as well. Thought it wasn't included initially.

File: client/lib/analytics/ad-tracking/index.js (hunk 2 above)
Comment (briowill) [reply client/lib/analytics/ad-tracking/index.js:4]: It should still be at the root level. This was an oversight on my side.

**Review by gmovr — APPROVED** [review]
I tested this again:
- [x] I checked out the branch locally, and allowed `ad-tracking` and `cookie-banner` in the `development.json` config file. 
- [x] Cookie banner works as expected
- [x] When tracking is allowed, all the events that are using `gaRecordEvent`, `recordGoogleEvent` etc. via /state/analytics are working. E.g. user engagement. 
- [x] No console or network errors
- [x] I did some gripping across the repo to ensure that nothing will miss the files/functions that are moved/deleted.
- [x] Tested purchases

@Automattic/jetpack-martech -- Could you take a look for Jetpack/Akismet?

**Review by robertsreberski — APPROVED** [review]
Code looks cleaner, and there are no changes in how GA4 for Jetpack/Akismet works. Thanks for the clean up!

**Pull request merged**